Modules/_testcapi/float.c

```diff
@@ -157,9 +157,41 @@ test_string_to_double(PyObject *self, PyObject *Py_UNUSED(ignored))
 }
 
 
+/*[clinic input]
+_testcapi.float_set_snan
+
+    obj: object
+    /
+
+Make a signaling NaN.
+[clinic start generated code]*/
+
+static PyObject *
+_testcapi_float_set_snan(PyObject *module, PyObject *obj)
+/*[clinic end generated code: output=f43778a70f60aa4b input=c1269b0f88ef27ac]*/
+{
+    if (!PyFloat_Check(obj)) {
+        PyErr_SetString(PyExc_ValueError, "float-point number expected");
+        return NULL;
+    }
+    PyObject *ret = PyNumber_Positive(obj);
+    if (!ret) {
+        return NULL;
+    }
+    double *d = &((PyFloatObject *)ret)->ob_fval;
+    if (!isnan(*d)) {
+        PyErr_SetString(PyExc_ValueError, "nan expected");
+        return NULL;
+    }
+    uint64_t *v = (uint64_t *)d;
```

Comment:
Would it be possible to use either memcpy() or an union to avoid strict aliasing issues?